### ISSUE TITLE
Support multiple values per field

### ISSUE THREAD

**Issue by indigoxela** [backdrop-contrib/resource_timeslots#2]
Time to fix this. It requires a lot of rewrite, but should be straight forward.

**Comment by indigoxela** [comment]
Is anyone willing to test this? Especially someone with Safari (the timezone smashing browser). :wink:

**Comment by indigoxela** [comment]
Here's a screenshot:

![screenshot-resource-timeslot](https://user-images.githubusercontent.com/842066/136693892-429c250e-f294-49f7-b148-b5569d239a43.png)

**Comment by yorkshire-pudding** [comment]
@indigoxela - I'll have a go at testing.  I don't have Safari though so can't test that.

**Comment by yorkshire-pudding** [comment]
@indigoxela worked well on desktop.  couldn't get any appointments to actually select on mobile (Android, Chrome).  Note -  I hadn't tested on mobile before this branch so not sure if this  breaks or not.

On mobile, the slot shows as selected but doesn't add the blue appointment.  When try to submit, get validation for no appointment time (name of field on my test).

**Comment by indigoxela** [comment]
Hi @yorkshire-pudding many thanks for testing!

Re touch devices: Fullcalendar does support them. To add an appointment, you need to tap longer (almost a second), then you can add. To resize it, you need to tap again (long), then you get a draggable resizer. To "unblock" that item for deletion, you need to tap once outside the slot, then the small delete button will work again.

> On mobile, the slot shows as selected but doesn't add the blue appointment

Hm, I'm afraid, I don't understand that. When does it show as selected? When opening the node form again? And what does "get validation for no appointment..." mean? That it validates or that it's considered empty?

Can you add some info about the OS version and browser on your mobile device?

**Comment by yorkshire-pudding** [comment]
> Hm, I'm afraid, I don't understand that. When does it show as selected? When opening the node form again? And what does "get validation for no appointment..." mean? That it validates or that it's considered empty?

Sorry for not being clear enough.  When I "click" on slot with finger, the slot indicates being selected through the left and right borders of the cell disappearing (but also the left/right borders of all cells to the right disappear).  I wasn't sure if it had selected a slot but just wasn't showing so tried to save.  I got validation as I hadn't selected a cell so the field was empty.

>   Re touch devices: Fullcalendar does support them. To add an appointment, you need to tap longer (almost a second), then you can add. To resize it, you need to tap again (long), then you get a draggable resizer. To "unblock" that item for deletion, you need to tap once outside the slot, then the small delete button will work again.

This works as you described.  Not the most intuitive UX though.

> Can you add some info about the OS version and browser on your mobile device?

Android version 9
Chrome version 94.0.04606.71

**Comment by indigoxela** [comment]
> This works as you described. Not the most intuitive UX though.

Great that it works! Regarding UX ... we probably can't do much about it - that's how Fullcalendar does it. It's possible to make the required tap time shorter, though (possibly with side effects). Maybe open another issue for that?

**Comment by indigoxela** [comment]
@yorkshire-pudding a quick thought regarding the disappearing borders - let me guess: you're using Seven. In Seven Fullcalendar looks awkward, because that theme's table styles interfere so much. Try to use another theme for content creation / editing.

**Comment by indigoxela** [comment]
Thank you both for testing (it has been confirmed that it also works in Safari in the chat), merged.

**Comment by olafgrabienski** [comment]
> To "unblock" that item for deletion, you need to tap once outside the slot, then the small delete button will work again.

I guess deleting doesn't work for me on a touch device. Before opening a new issue, I'm trying to clarify how it's supposed to work: Is "the small delete button" the same as on other devices, i.e. the cross at the top right corner of the item?

**Comment by indigoxela** [comment]
> I'm trying to clarify how it's supposed to work: Is "the small delete button" the same as on other devices, i.e. the cross at the top right corner of the item?

Yes, exactly that.
Hm... seems broken for me, too now. :disappointed: 

When tapping on a selected slot a little longer, a small "dot" appears at the bottom of the slot:

![blocked-dot](https://user-images.githubusercontent.com/842066/145838202-5d3ed144-6acf-44da-949b-af3ae5fb1fc7.png)

With that dot it's possible to resize that slot or drag it around on touch devices.

The trick to get the delete button event back, was to tap short outside the selection (dot vanishes). Then the delete button (cross) worked again. But that's not working anymore.

Bad news: this means, the event isn't bubbling through anymore. There's probably nothing I can do about that.

**Comment by indigoxela** [comment]
@olafgrabienski please open a new issue for that, because that's unrelated to multi-value.
